### PR TITLE
docs: remove remaining bluebird references

### PR DESCRIPTION
**Pull request by SimonSchick**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

- Removes bluebird references from docs
- Removes bluebird typings
- Cleans up some unused imports in typings where bluebird was removed

### CODE REVIEW

File: docs/manual/other-topics/typescript.md [hunk 1]
```diff
@@ -129,7 +128,6 @@ Address.belongsTo(User, {targetKey: 'id'});
 User.hasOne(Address,{sourceKey: 'id'});
 
 async function stuff() {
-  // Please note that when using async/await you lose the `bluebird` promise context
   // and you fall back to native
```
Comment (kroleg) [inline docs/manual/other-topics/typescript.md:131]: i think that this line needs to be removed too

**Comment by SimonSchick** [comment]
Oh look the flaky row locking test again >.>

**Comment by sushantdhiman** [comment]
> Oh look the flaky row locking test again >.>

Yes annoying

File: types/test/e2e/docs-example.ts [hunk 2]
```diff
@@ -120,7 +120,6 @@ Address.belongsTo(User, {targetKey: 'id'});
 User.hasOne(Address,{sourceKey: 'id'});
 
 async function stuff() {
-  // Please note that when using async/await you lose the `bluebird` promise context
   // and you fall back to native
```
Comment (kroleg) [inline types/test/e2e/docs-example.ts:123]: and this comment line should be removed too

**Pull request merged**